app/src/main/java/com/sopt/gongbaek/presentation/ui/mypage/screen/MyPageScreen.kt

```diff
@@ -1,32 +1,238 @@
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
-import androidx.compose.ui.unit.sp
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.LocalLifecycleOwner
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.flowWithLifecycle
+import com.sopt.gongbaek.R
+import com.sopt.gongbaek.presentation.model.ProfileImageList
+import com.sopt.gongbaek.presentation.type.MyGroupPagerType
+import com.sopt.gongbaek.presentation.ui.component.tabpager.CustomTabPager
+import com.sopt.gongbaek.presentation.ui.component.topbar.CenterTitleTopBar
+import com.sopt.gongbaek.presentation.ui.mypage.component.MyGroupScreenContent
+import com.sopt.gongbaek.presentation.util.extension.roundedBackgroundWithBorder
+import com.sopt.gongbaek.ui.theme.GongBaekTheme
 
 @Composable
-fun MyPageRoute() {
-    MyPageScreen()
+fun MyPageRoute(
+    viewModel: MyPageViewModel = hiltViewModel(),
+    navigateGroupDetail: (Int, String) -> Unit,
+    navigateGroupRoom: (Int, String) -> Unit
+) {
+    val myPageUiState by viewModel.state.collectAsStateWithLifecycle()
+    val lifecycleOwner = LocalLifecycleOwner.current
+    val myPageTabs: List<String> = MyGroupPagerType.entries.map { it.description }
+    val pagerState = rememberPagerState { myPageTabs.size }
+
+    LaunchedEffect(pagerState.currentPage) {
+        when (pagerState.currentPage) {
+            0 -> viewModel.setEvent(MyPageContract.Event.OnRegisterGroupsTabClick)
+            1 -> viewModel.setEvent(MyPageContract.Event.OnApplyGroupsTabClick)
+        }
+    }
+
+    LaunchedEffect(viewModel.sideEffect, lifecycleOwner) {
+        viewModel.sideEffect.flowWithLifecycle(lifecycleOwner.lifecycle)
+            .collect { sideEffect ->
+                when (sideEffect) {
+                    is MyPageContract.SideEffect.NavigateGroupDetail -> {
+                        navigateGroupDetail(sideEffect.groupId, sideEffect.groupCycle)
+                    }
+
+                    is MyPageContract.SideEffect.NavigateGroupRoom -> {
+                        navigateGroupRoom(sideEffect.groupId, sideEffect.groupCycle)
+                    }
+                }
+            }
+    }
+
+    MyPageScreen(
+        uiState = myPageUiState,
+        myPageTabs = myPageTabs,
+        pagerState = pagerState,
+        onGroupDetailButtonClick = { groupId, groupCycle -> viewModel.sendSideEffect(MyPageContract.SideEffect.NavigateGroupDetail(groupId, groupCycle)) },
+        onGroupRoomButtonClick = { groupId, groupCycle -> viewModel.sendSideEffect(MyPageContract.SideEffect.NavigateGroupRoom(groupId, groupCycle)) }
+    )
+}
+
+@Composable
+fun MyPageScreen(
```

Comment:
[2] 외부에서 접근이 불가능하게 가시성을 제한하는건 어떤지 의견을 남겨봅니다~!
```suggestion
private fun MyPageScreen(
```